tests/common/utils.py

```diff
@@ -66,3 +69,22 @@ def inverted_test(*arguments, **kwargs):
 
 
 fails = fails_with(AssertionError)
+
+
+def checks_deprecated_behaviour(func):
```

Comment:
It would be nice to see a couple of usages of it so that it's (implicitly) tested. Would you mind taking a test for deprecated behaviour or two and converting it to use of this decorator?